### PR TITLE
New Schemes Ready

### PR DESCRIPTION
**Pull request by DarianHole**
Here are the TB and Typhi schemes

I've added an automatic metadata inclusion for the typhi scheme that can also take an additional metadata table. It is currently under main as I couldn't figure out how to get it to work under the const.py script. I can change that if needed.

Still to do:
~~- Create pytests for these new schemes~~
~~- Put in the full tb scheme, it is currently a placeholder as I wait for the new one~~
~- Test the tb scheme~

TB scheme works well and is accessed under -s "tb_speciation" as requested

Pytests may take me some time as I have not done them before but I'll get those up soonish hopefully.

### CODE REVIEW

File: bio_hansel/main.py [hunk 1]
```diff
@@ -200,6 +201,11 @@ def main():
     df_md = None
     if args.scheme_metadata:
         df_md = read_metadata_table(args.scheme_metadata)
+    if scheme == 'typhi':
```
Comment (peterk87) [inline bio_hansel/main.py]: I like this approach to integrating existing scheme metadata into the subtyping metadata dataframe. Would it be possible to extend this to all built-in typing schemes? I'm not sure if `resource_filename` would play nice with missing files, but you might be able to wrap it in a `try/except` block:

```python
try:
    ...
    df_md = pd.concat([df_md, read_metadata_table(resource_filename(program_name, f'data/{scheme}/{scheme}_scheme_metadata.tsv'))], axis=1)
    ...
except Exception: # probably a better idea to catch a ResourceNotFoundException or whatever it is
    pass
```

File: tests/test_subtyping_processing.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 from bio_hansel.subtype import Subtype
 from bio_hansel.subtyper import absent_downstream_subtypes, sorted_subtype_ints, count_periods, empty_results
 from bio_hansel.utils import find_inconsistent_subtypes
+from bio_hansel.subtype_stats import SubtypeCounts
```
Comment (peterk87) [inline tests/test_subtyping_processing.py]: Is `SubtypeCounts` used in this file?

**Comment by dankein** [comment]
Hi @peterk87 yes we'll need a metadata file to make the scheme hierarchical again (I still need to make this due to some recent changes, hopefully very shortly)

**Comment by glabbe** [comment]
@peterk87 We will want to add metadata to the SE scheme after we decide on a shorter nomenclature targeted at source attribution, in collaboration with the epis and the NML surveillance team. It's not ready yet, but it's coming...

File: tests/test_subtyping_processing.py (hunk 2 above)
Comment (DarianHole) [reply tests/test_subtyping_processing.py]: I believe that was a mistype that I forgot to correct

**Comment by DarianHole** [comment]
Merging in the new development to get fix incompatibilities should any be found.
    - Looks like it

File: bio_hansel/main.py (hunk 1 above)
Comment (DarianHole) [reply bio_hansel/main.py]: Ya I'll put that on my list of things to do and get to it. I think that is a much more elegant solution then including the case just for typhi as I currently have it

File: tests/test_subtyping_processing.py (hunk 2 above)
Comment (DarianHole) [reply tests/test_subtyping_processing.py]: It is not used in this commit but it is used to test the regex in a previous one and it was added here early by mistake so it should be there I guess

**Comment by DarianHole** [comment]
With the following change, metadata can either be specified on the command line, or included with the code (as typhi currently is). This allows for the following 4 cases of metadata to work with the program:

- No metadata included and none specified
- Metadata included with code added and none specified
- No metadata included and specified metadata added
- Metadata included with the code is added and metadata specified is added and then they are combined together and combined with the results

I believe that this is the best way to allow this to go through. Let me know if anything else should be changed with it.

File: bio_hansel/main.py (hunk 1 above)
Comment (DarianHole) [reply bio_hansel/main.py]: Done with commit 6a5e0e4, it now should work with any added files should they be called "metadata.tsv". 

I had to create the file path with some basic string concatenation as I couldn't get it to work with the {}.

Comment (peterk87) [reply bio_hansel/main.py]: You need the `f` before the string to perform string literal interpolation `f'data/{scheme}/metadata.tsv'` where `scheme` is a variable that exists in that scope. (see https://www.python.org/dev/peps/pep-0498/ for more info). It's a Python 3.6+ feature as an alternative to `'data/{scheme}/metadata.tsv'.format(scheme=scheme)` or using `+` to build up your strings.

**Comment by peterk87** [comment]
@DarianHole I think that's great that you're building in so much flexibility with the subtype metadata. It'll make biohansel more useful for a larger audience :+1:

File: bio_hansel/main.py (hunk 1 above)
Comment (DarianHole) [reply bio_hansel/main.py]: Oh I see, I missed that. I'll give that a go as it looks way better then what I have

**Pull request merged**